lib/url.c

```diff
@@ -4049,9 +4051,31 @@ static CURLcode parseurlandfillconn(struct Curl_easy *data,
    * proxy -- and we don't know if we will need to use SSL until we parse the
    * url ...
    ************************************************************/
-  if((2 == sscanf(data->change.url, "%15[^:]:%[^\n]",
-                  protobuf, path)) &&
-     strcasecompare(protobuf, "file")) {
+  for(i = 0; i < 16 && data->change.url[i] &&
+             data->change.url[i] != '\n'; ++i) {
```

Comment:
This function already verifies that there is no CR or LF in the URL, immediately before this code snippet so this newline check is superfluous.